micro-infra-camel/src/main/java/com/ofg/infrastructure/camel/CorrelationIdInterceptor.java

```diff
@@ -0,0 +1,59 @@
+package com.ofg.infrastructure.camel;
+
+import com.ofg.infrastructure.correlationid.CorrelationIdUpdater;
+import com.ofg.infrastructure.correlationid.UuidGenerator;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+import static com.ofg.infrastructure.correlationid.CorrelationIdHolder.CORRELATION_ID_HEADER;
+
+/**
+ * Interceptor class that ensures the correlationId header is present in {@Exchange}.
+ */
+public class CorrelationIdInterceptor implements Processor {
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final UuidGenerator uuidGenerator;
+
+    public CorrelationIdInterceptor(UuidGenerator uuidGenerator) {
+        this.uuidGenerator = uuidGenerator;
+    }
+
+    /**
+     * Ensures correlationId header is set in incoming message (if is missing a new correlationId is created and set).
+     *
+     * @param exchange Camel's container holding received message
+     * @throws Exception if an internal processing error has occurred
+     */
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        String correlationIdHeader = getCorrelationId(exchange);
+        CorrelationIdUpdater.updateCorrelationId(correlationIdHeader);
+        setCorrelationIdHeaderIfMissing(exchange, correlationIdHeader);
+    }
+
+    private String getCorrelationId(Exchange exchange) {
```

Comment:
getCorrelationId and setCorrelationIdHeaderIfMissing methods have inconsistent names.